rules/no-empty-file.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+const MESSAGE_ID = 'no-empty-file';
+const messages = {
+	[MESSAGE_ID]: 'Empty files are not allowed.',
+};
+
+const isEmpty = node =>
+	node.type === 'EmptyStatement'
+	|| (node.type === 'ExpressionStatement' && node.expression.value === 'use strict')
```

Comment:
This is wrong, `"use asm"` is also a directive, while `("use strict")` is not.

Reply:
I am not sure what you want me to change here. Are you wanting me to check for `.directive` instead of the expression value?

Reply:
Yes. That's correct way to check directives.

Reply:
I can check `.directive` if `'use strict` is at the top level, however `{'use strict';}` doesn't have a `.directive`, which is why I started checking it this way instead.

Reply:
Sorry about this wrong example

```js
{
	'use strict';
}
```

it's not a directive, should be valid.

Reply:
@fisker that means this should be valid too, `;;'use strict';`, correct?

Reply:
This should be fixed.